fabtests/multinode/src/harness.c

```diff
@@ -234,11 +273,19 @@ static int pm_conn_setup(void)
 	}
 #endif
 
-	ret = bind(sock, (struct sockaddr *)&pm_job.oob_server_addr,
-		  pm_job.server_addr_len);
-	if (ret == 0) {
-		ret = server_connect();
-	} else {
+	if (!sched || (sched && pm_job.my_rank == 0)) {
+		/* Rank 0 should be the server and the only one that should call
+		* bind() and server_connect() */
+		ret = bind(sock, (struct sockaddr *)&pm_job.oob_server_addr,
+			pm_job.server_addr_len);
+		if (ret == 0) {
+			ret = server_connect();
+			bound = true;
+		} else if (sched) {
+			ret = -FI_EINVAL;
```

Comment:
Why overwrite the error code? The original code could be more helpful in diagnosing the connection error.